### DISCUSSION TITLE
Recipe for collection.toArray(new Foo[collection.size()])

### DISCUSSION THREAD

**Discussion by Bananeweizen** [openrewrite/rewrite#3590] [Ideas]
When converting collections to arrays, many people still write `collection.toArray(new Foo[collection.size()])` in an attempt to optimize the allocation of the new array. However, using `...new Foo[0]` is not slower on current JVMs, and more easy to read. There should be some recipe to implement this cleanup.

Detailed technical background: https://shipilev.net/blog/2016/arrays-wisdom-ancients/

(And I'm not sure if I should instead raise a feature request, not a discussion topic)

**Comment by timtebeek** [comment]
Hi! Thanks for the suggestion (and other activities)! Fully agree with this suggestion; I'd logged that as a recipe suggestion here:
- https://github.com/openrewrite/rewrite-static-analysis/issues/84

If you want an easy way to search across issues for all OpenRewrite projects then I recommend:
https://github.com/orgs/openrewrite/projects/4/views/10

Hope that helps! We can continue the the discussion around this recipe in https://github.com/openrewrite/rewrite-static-analysis/issues/84
